pkg/services/baremetal/host/host.go

```diff
@@ -1202,6 +1205,131 @@ func (s *Service) actionImageInstalling(ctx context.Context) actionResult {
+
+	case sshclient.ImageURLCommandStateFailed:
+		record.Warn(s.scope.HetznerBareMetalHost, "InstallImageNotSuccessful", logFile)
+		msg := "image-url-command failed"
+		s.scope.Logger.Error(nil, msg, "logFile", logFile)
+		conditions.MarkFalse(host, infrav1.ProvisionSucceededCondition,
+			"ImageURLCommandFailed", clusterv1.ConditionSeverityWarning,
+			"%s", msg)
+		return s.recordActionFailure(infrav1.FatalError, msg)
+
+	case sshclient.ImageURLCommandStateNotStarted:
+		data, err := s.scope.GetRawBootstrapData(ctx)
+		if err != nil {
+			return actionError{err: fmt.Errorf("baremetal GetRawBootstrapData failed: %w", err)}
+		}
+		if s.scope.ImageURLCommand == "" {
+			err = errors.New("internal error: --baremetal-image-url-command is not set?")
+			s.scope.Logger.Error(err, "")
+			conditions.MarkFalse(s.scope.HetznerBareMetalHost, infrav1.ProvisionSucceededCondition,
+				"ImageURLCommandMissing",
+				clusterv1.ConditionSeverityError,
+				"%s", err.Error())
+			return actionError{err: err}
+		}
+		// get the information about storage devices again to have the latest names.
+		// Device names can change during restart.
+		storage, err := obtainHardwareDetailsStorage(sshClient)
+		if err != nil {
+			return actionError{err: fmt.Errorf("failed to obtain hardware details storage: %w", err)}
+		}
+
+		// get device names from storage device
+		deviceNames := getDeviceNames(s.scope.HetznerBareMetalHost.Spec.RootDeviceHints.ListOfWWN(), storage)
+
+		exitStatus, stdoutStderr, err := sshClient.StartImageURLCommand(ctx, s.scope.ImageURLCommand, s.scope.HetznerBareMetalHost.Spec.Status.InstallImage.Image.URL, data, s.scope.Hostname(), deviceNames)
+		if err != nil {
+			err := fmt.Errorf("StartImageURLCommand failed (retrying): %w", err)
+			// This could be a temporary network error. Retry.
+			s.scope.Logger.Error(err, "",
+				"ImageURLCommand", s.scope.ImageURLCommand,
+				"exitStatus", exitStatus,
+				"stdoutStderr", stdoutStderr)
+			conditions.MarkFalse(s.scope.HetznerBareMetalHost, infrav1.ProvisionSucceededCondition,
+				"ImageURLCommandFailedToStart",
+				clusterv1.ConditionSeverityWarning,
+				"%s", err.Error())
+			return actionError{err: err}
+		}
+
+		if exitStatus != 0 {
+			msg := "StartImageURLCommand failed with non-zero exit status. Deleting machine"
+			s.scope.Logger.Error(nil, msg,
+				"ImageURLCommand", s.scope.ImageURLCommand,
+				"exitStatus", exitStatus,
+				"stdoutStderr", stdoutStderr)
+			conditions.MarkFalse(s.scope.HetznerBareMetalHost, infrav1.ProvisionSucceededCondition,
+				"StartImageURLCommandFailed",
+				clusterv1.ConditionSeverityWarning,
+				"%s", msg)
+			return s.recordActionFailure(infrav1.ProvisioningError, msg)
+		}
+
+		conditions.MarkFalse(s.scope.HetznerBareMetalHost, infrav1.ProvisionSucceededCondition,
+			"ImageURLCommandStared",
+			clusterv1.ConditionSeverityInfo,
+			"baremetal-image-url-command started")
+
+		record.Event(s.scope.HetznerBareMetalHost, "ExecuteImageURLCommand",
```

Comment:
I believe that you should create an event maybe in the cases where this fails, and maybe when it finishes, but why creating one when you start the command?!